app/src/main/java/space/taran/arkshelf/data/LinkLocalDataSource.kt

```diff
@@ -11,17 +20,51 @@ import kotlin.io.path.outputStream
 import kotlin.io.path.writeText
 import kotlin.io.path.createTempFile
 import java.security.MessageDigest
+import java.util.zip.ZipFile
+import javax.inject.Inject
+import kotlin.io.path.extension
+import kotlin.io.path.getLastModifiedTime
+import kotlin.io.path.isDirectory
 
-
-class LinkLocalDataSource() {
+class LinkLocalDataSource @Inject constructor(
+    private val preferences: UserPreferences
+) {
     private val klaxon = Klaxon()
+    private val latest = mutableMapOf<Link, Path>()
+    private val loadMoreMutex = Mutex()
 
     fun createLinkFile(link: Link, basePath: Path) {
         val jsonFile = generateJsonFile(link)
         val savePath = basePath.resolve(Path("${sha512(link.url)}.link"))
         zipFiles(jsonFile, link.imagePath, savePath)
     }
 
+    suspend fun loadMore(): List<Link> = withContextAndLock(
+        Dispatchers.IO,
+        loadMoreMutex
+    ) {
+        val linkFolder = preferences.getLinkFolder()
+            ?: return@withContextAndLock latest.keys.toList()
+
+
+        val newLinks = linkFolder
+            .listChildren()
+            .filter {
+                it.extension == "link" && !it.isDirectory()
+            }
+            .sortedByDescending { it.getLastModifiedTime() }
+            .minus(latest.values)
+            .take(CHUNK_SIZE)
```

Comment:
Maybe "batch" or "page" would be better word. "Chunk" sounds like part of a binary file to me.

Reply:
Renamed to "batch"